File: cpp/reverse-string/reverse_string.cpp
#include "reverse_string.h"

using namespace std;

namespace reverse_string
{
  string reverse_string(string s) {
    string res = "";
    if (s.length() == 0) {
      return res;
      };
    for (int i = s.length() - 1; i >= 0; i--) {
      res += s[i];
    }
    return res;
  }
}  // namespace reverse_string

File: nim/reverse-string/reverse_string.nim

proc reverse*(s: string): string =
  var res = ""
  let l = len(s)
  if l == 0: return res
  for i in countdown(l-1,0):
      res = res & $s[i]
  return res

#[
for c in s:
    result = c & result
]#